### PR TITLE
feat: Added Authentication

### PR DESCRIPTION
**Pull request by mnath30**
Added the following:
1. Login
2. Signup
3. Logout

### CODE REVIEW

File: src/components/Navigation/Navigation.js [hunk 1]
```diff
@@ -26,9 +30,20 @@ const Navigation = ({ setShowMobileNav }) => {
...
-        </button>
+        {isLoggedIn ? (
+          <Link to="/logout">
+            <button className="secondary__btn ">
+              <i className="fa-regular fa-user secondary__btn-icon"></i>LOG OUT
+            </button>
+          </Link>
+        ) : (
+          <Link to="/login">
+            {" "}
```
Comment (Prabhav-Sharma) [inline src/components/Navigation/Navigation.js]: Remove these, it's annoying how it gets added. Unless you added them yourself. 😅

File: src/pages/Authentication/Signup.jsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+import { fetchLoginDetails } from "../../services/fetchAuth";
+import { useAuth } from "../../context";
+import { LOADING, SIGNUP, ERROR } from "../../helper/constants";
+import { Loader } from "../../components";
+
+const Signup = () => {
+  const navigate = useNavigate();
+  const { authState, authDispatch } = useAuth();
+  const { loading, error } = authState;
+  const [userName, setUserName] = useState("");
```
Comment (Prabhav-Sharma) [inline src/pages/Authentication/Signup.jsx:13]: Suggestion: Can create a reducer to manage all form states in both pages

**Review by Prabhav-Sharma — APPROVED** [review]
The code is clean and follows best practices, consider the suggestions I've given, regardless of those, PR is good to be merged

File: src/services/fetchAuth.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import axios from "axios";
+
+const fetchLoginDetails = async (userDetails, login = false) => {
+  const apiResponse = { token: "", username: "", error: "" };
+  try {
+    let response;
+    if (login) {
+      response = await axios.post("/api/auth/login", userDetails);
```
Comment (rehanmohiuddin) [inline src/services/fetchAuth.js]: make these seperate services for login and sign up as the code scales and you want to add emaill / otp verification it becomes difficult just a suggestion

**Review by rehanmohiuddin — APPROVED** [review]

File: src/components/Navigation/Navigation.js (hunk 1 above)
Comment (mnath30) [reply src/components/Navigation/Navigation.js]: Removed this

File: src/pages/Authentication/Signup.jsx (hunk 2 above)
Comment (mnath30) [reply src/pages/Authentication/Signup.jsx:13]: Thankyou Prabhav for this suggestion. Will keep this in mind and change this after completing my P0 features.

File: src/services/fetchAuth.js (hunk 3 above)
Comment (mnath30) [reply src/services/fetchAuth.js]: Thankyou for your suggestion. Have made the changes

**Pull request merged**